DBZ-6721 Add docs for SMTs (#5634)
debezium/debezium: documentation/modules/ROOT/pages/transformations/vitess-use-local-vgtid.adoc

:page-aliases: configuration/vitess-use-local-vgtid.adoc
[id="vitess-use-local-vgtid"]
= Vitess Use Local VGTID

:toc:
:toc-placement: macro
:linkattrs:
:icons: font
:source-highlighter: highlight.js

toc::[]

The vitess-specific `UseLocalVgtid` SMT is used to transform the VGTID from a list of several shards VGTIDs to a list of a single shard's VGTID for the shard that received the change for this event.

It is done by using the `shard` field of the `source` of the event to determine the shard that received the change. Then that shard value is used to reduce the VGTID to only the relevant shard.

> 

Reviewer: ```suggestion
The following example shows how to add the SMT to the connector configuration so that it converts the list of VGTIDs to the value of the local VGTID.
```

[[example-vitess-use-local-vgtid.adoc]]

> == Example

Reviewer: ```suggestion
== Example: Configuring the `UseLocalVgtid` SMT
```

> To transform VGTIDs to local VGTIDs, use the following configuration:

Reviewer: ```suggestion

```
Ok to remove if we convey the information in the Example header and intro sentence.

:page-aliases: configuration/vitess-use-local-vgtid.adoc
[id="vitess-use-local-vgtid"]
= Vitess Use Local VGTID

:toc:
:toc-placement: macro
:linkattrs:
:icons: font
:source-highlighter: highlight.js

toc::[]

The `UseLocalVgtid` SMT converts the value of the VGTID in the event record from a list of VGTIDs for several shards to the VGTID for the single shard that received the change for this event.
This transformation is designed for use only with the {prodname} connector for Vitess.

The SMT extracts the `shard` field in the `source` block of the transaction boundary event record to determine the shard that received the change. 

> The SMT then sets that value of the VGTID in the even record to the value of the VGTID for the specified shard.

Reviewer: ```suggestion
The SMT then sets that value of the VGTID in the event record to the value of the VGTID for the specified shard.
```

Other VGTID values are removed from the event record.

> The following example shows how to add the SMT to the connector configuration so that it converts the list of VGTIDs to the value of the local VGTID.

Reviewer: ```suggestion
The following example shows how to add the SMT to the connector configuration so that it converts lists of the VGTIDs for multiple shards to the value of a single local VGTID.
```
